### PR TITLE
Use Dockerfile to make development and deployment easier

### LINKED ISSUE #37: Better and more useful README
The current README doesn't help new contributors set everything up

### LINKED ISSUE #115: Add Docker container for `rq worker`
I want to make redis less annoying in development. Right now you have to run a `tmux` session or something with the `rq worker` command going, but there are containers for that which we could chuck in the `docker-compose.yml`.

### PR DESCRIPTION
**Pull request by WillNilges**
Hot damn.

Closes #37 
Closes #115

This PR finally does away with S2I on OKD (It's currently pulling from this branch on Prod!) by setting up a Dockerfile for this project that can be used on `proxmox` and both `rq` pods. Instructions are included for setting everything up as well.

This also adds the `large.sql` schema back to the repo.

### CODE REVIEW

**Review by galenguyer — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.8-buster
+WORKDIR /opt/proxstar
+RUN apt-get update -y && apt-get install -y python3-dev libldap2-dev libsasl2-dev ldap-utils tox lcov valgrind
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+COPY start_worker.sh start_scheduler.sh .
+COPY .git ./.git # TODO (willnilges): The app crashes without it. Remove if you can.
```
Comment (mxmeinhold) [inline Dockerfile:7]: This is needed for the version tag at the bottom of the page

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 black~=21.9b0
-csh-ldap~=2.3.1
+# csh-ldap~=2.3.1
```
Comment (devinmatte) [inline requirements.txt:2]: You should remove this comment

File: HACKING/.env.template [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+PROXSTAR_PROXMOX_VM_STORAGE=ceph
+PROXSTAR_PROXMOX_SSH_USER=root
+PROXSTAR_PROXMOX_SSH_KEY="" # Ask an RTP. This is gonna look like a certificate.
+PROXSTAR_PROXMOX_SSH_KEY_PASS= # Password for above certificate
+
+# STARRS
+PROXSTAR_STARRS_DB_HOST=proxstar-postgres
+PROXSTAR_DB_NAME=starrs
+PROXSTAR_DB_USER=postgres
+PROXSTAR_DB_PASS=tits12347
```
Comment (devinmatte) [inline HACKING/.env.template:37]: Please don't publish this is in public open source repo

File: Dockerfile (hunk 1 above)
Comment (WillNilges) [reply Dockerfile:7]: Is there a specific file i can `COPY` without having to do the entire directory?

File: HACKING/.env.template (hunk 3 above)
Comment (WillNilges) [reply HACKING/.env.template:37]: These are credentials for the development DB container.

Comment (devinmatte) [reply HACKING/.env.template:37]: Then change the password

**Comment by WillNilges** [comment]
For some reason, proxstar doesn't want to exit cleanly from its container. Whenever I do `podman stop proxstar` it hangs and has to resort to killing it. It happens for proxtar, and both rq containers.

`WARN[0010] StopSignal SIGTERM failed to stop container proxstar-rq in 10 seconds, resorting to SIGKILL`

**Comment by WillNilges** [comment]
This works OK for local development, but I'm not sure it will work for OKD.

**Comment by WillNilges** [comment]
There's no way to check this on the build logs 🤦🏻 I'm tempted to add another layer that just echos the environment variable.

**Comment by WillNilges** [comment]
yeah this sucks.

**Comment by WillNilges** [comment]
https://artsy.github.io/blog/2018/09/10/Dockerhub-Stamping-Commits/

**Comment by WillNilges** [comment]
https://docs.openshift.com/container-platform/3.4/dev_guide/builds/build_hooks.html

**Comment by WillNilges** [comment]
https://github.com/openshift/origin/issues/615

**Pull request merged**